components/brave_news/browser/channels_controller.cc

```diff
@@ -123,11 +127,25 @@ std::vector<std::string> ChannelsController::GetChannelLocales(
 
 void ChannelsController::GetAllChannels(ChannelsCallback callback) {
   publishers_controller_->GetOrFetchPublishers(base::BindOnce(
-      [](ChannelsCallback callback, PrefService* prefs, Publishers publishers) {
+      [](base::WeakPtr<ChannelsController> controller,
```

Comment:
if you pass it like this, you have to check if the weak ptr is valid, otherwise there is no sense in doing that